sphinx/builders/gettext.py

```diff
@@ -166,6 +166,20 @@ def dst(self, dt):
 ltz = LocalTimeZone()
 
 
+def should_write(filepath, new_content):
+    if not path.exists(filepath):
+        return True
+    with open(filepath, 'r', encoding='utf-8') as oldpot:
+        old_content = oldpot.read()
+        old_header_index = old_content.index('"POT-Creation-Date:')
+        new_header_index = old_content.index('"POT-Creation-Date:')
+        old_body_index = old_content.index('"PO-Revision-Date:')
+        new_body_index = new_content.index('"PO-Revision-Date:')
+        return ((old_content[:old_header_index] != new_content[:new_header_index]) or
+                (new_content[new_body_index:] != old_content[old_body_index:]))
```

Comment:
Indeed, our template for .po file: `POHEADER` has fixed `PO-Revision-Date`. It's immutable.
```
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
```
And current time is used for `POT-Creation-Date:`. so `should_write()` would return True.

So we have to compare without these headers.